test/cli_test.js

```diff
@@ -192,22 +193,24 @@ describe('cli', function() {
     });
 
     context('with param --environment', function() {
-      it('defaults to --environment=development: { env: "development" }', function() {
+      it('defaults to { env: "development" } and sets process.env.BROCCOLI_ENV = "development"', function() {
         const spy = sinon.spy(loadBrocfile());
         sinon.stub(broccoli, 'loadBrocfile').value(() => spy);
 
-        return cli(['node', 'broccoli', 'build', 'dist']).then(() =>
-          chai.expect(spy).to.be.calledWith(sinon.match.has('env', 'development'))
-        );
+        return cli(['node', 'broccoli', 'build', 'dist']).then(() => {
+          chai.expect(spy).to.be.calledWith(sinon.match.has('env', 'development'));
+          chai.expect(process.env.BROCCOLI_ENV).to.be.undefined;
```

Comment:
I do not think this is _ever_ expected...

Reply:
This test is confirming that if `--environment` is not set, then `BROCCOLI_ENV` does not get set. This maintains the existing behavior of `broccoli build`, that `BROCCOLI_ENV` would be undefined if you don't manually define it. Doing `--environment=production` is basically an alias for `BROCCOLI_ENV=production broccoli build`